osf/management/commands/withdraw_all_preprints_from_provider.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from django.core.management.base import BaseCommand
+
+from osf.models import PreprintProvider, PreprintRequest, OSFUser
+from osf.utils.workflows import DefaultStates, RequestTypes
+
+PAGE_SIZE = 100
+
+PROVIDER_WITHDRAWAL_COMMENT = {
+    'eartharxiv': 'This preprint was moved to the new EartharXiv hosted by CDL. The DOI now resolves to the new location at https://eartharxiv.org/.'
+}
+
+logger = logging.getLogger(__name__)
+
+
+def withdraw_all_preprints(provider_id, page_size, user_guid, comment=None):
+    """ A management command to withdraw all preprints from a specified provider
+        Created to withdraw all EarthRxiv preprints, but can be used for any preprint
+        provider
+    """
+    provider = PreprintProvider.load(provider_id)
+    if not provider:
+        raise RuntimeError('Could not find provider. Check ID and try again')
+
+    user = OSFUser.load(user_guid)
+    if not user:
+        raise RuntimeError('Could not find user. Check GUID and try again')
+
+    saved_comment = PROVIDER_WITHDRAWAL_COMMENT.get(provider_id, None)
+    if not comment and saved_comment:
+        comment = saved_comment
+
+    if not comment and not saved_comment:
+        raise RuntimeError('Comment not provided!')
+
+    preprints = provider.preprints.filter(date_withdrawn=None)[:page_size]
+
+    preprints_withdrawn = 0
+
+    for preprint in preprints:
+        preprint_request = PreprintRequest(
```

Comment:
Seeing `force_withdrawal` in the email template prompted me to dig a little. Calling `preprint.run_withdraw(user, comment)` is much simpler creating and walking through a request's approval, and also obfuscates the withdrawing user -- in my local test, it caused the `"A moderator has withdrawn your [...]"` email language, which seems preferable.